cli/tests/tui/test_executor.py

```
@@ -306,6 +342,102 @@ async def test_redirected_subprocess_output_is_decoded_as_utf8() -> None:
+
+    events = await _collect(executor, ("-c", "ignored"))
+
+    assert [event.text for event in events if event.kind == "output"] == ["€"]
+
+
+@pytest.mark.asyncio
+async def test_pipe_executor_flushes_unterminated_fragment_on_controlled_timeout(monkeypatch) -> None:
+    executor = _mock_pipe_executor(monkeypatch, (b"Select: ", b""))
+    wait_calls = 0
+
+    async def controlled_wait_for(awaitable, *, timeout: float):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `controlled_wait_for`

(ANN202)